src/libs/sfdk/virtualmachine.cpp

```diff
@@ -191,8 +186,16 @@ void VirtualMachine::setMemorySizeMb(int memorySizeMb, const QObject *context,
 
 int VirtualMachine::availableMemorySizeMb()
 {
-    QTC_CHECK(VirtualMachinePrivate::s_availableMemmorySizeMb > 0);
-    return VirtualMachinePrivate::s_availableMemmorySizeMb;
+#ifdef Q_OS_WIN
+    MEMORYSTATUSEX status;
+    status.dwLength = sizeof(status);
+    GlobalMemoryStatusEx(&status);
+    return status.ullTotalPhys >> 20;
+#else
+    long pages = sysconf(_SC_PHYS_PAGES);
+    long page_size = sysconf(_SC_PAGE_SIZE);
+    return (pages * page_size) >> 20;
```

Comment:
More warnings from clang: implicit conversion loses integer precision: 'long' to 'int'

I guess it would like you to make the cast explicit.

Reply:
I was surprised it didn't complain about these, so I thought it was so clever that >>20 made it quiet :) Unfortunately it's because -Wconversion is not included in -Wall and it is not possible to simply enable it with Qt used from build tree, as it issues a lot of warning from Qt headers. `QMAKE_CXXFLAGS += -isystem $$[QT_INSTALL_HEADERS]` does not help because when Qt is used from build tree, the headers are still in the source tree and so not covered by QT_INSTALL_HEADERS.